src/js/tech/ChromecastTech.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+var ChromecastSessionManager = require('../chromecast/ChromecastSessionManager'),
+    ChromecastTechUI = require('./ChromecastTechUI'),
+    _ = require('underscore'),
+    SESSION_TIMEOUT = 10 * 1000, // milliseconds
+    ChromecastTech;
+
+
+ChromecastTech = {
+   constructor: function(options) {
+      var subclass;
+
+      this._eventListeners = [];
+
+      // We have to initialize the UI here, before calling super.constructor
+      // because the constructor calls `createEl`, which references `this._ui`.
+      this._ui = new ChromecastTechUI();
+      this._ui.updatePoster(options.source.poster);
+
+      // Call the super class' constructor function
+      subclass = this.constructor.super_.apply(this, arguments);
+
+      this._chromecastSessionManager = options.source.chromecastSessionManager;
+      this._remotePlayer = this._chromecastSessionManager.getRemotePlayer();
+      this._remotePlayerController = this._chromecastSessionManager.getRemotePlayerController();
+      this._listenToPlayerControllerEvents();
+      this.on('dispose', this._removeAllEventListeners.bind(this));
+
+      this._hasPlayedAnyItem = false;
+      this._currentSource = options.source;
+      this._requestTitle = options.requestTitleFn || _.noop;
+      this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      // See `currentTime` function
+      this._firstItemStartTime = options.startTime || 0;
```

Comment:
"First item" lends itself to implying there are more items, i.e. a playlist. A tech only knows about a single source at a given time, i.e. one item. Therefore, if a tech only ever knows about one item, it could be said that every item is the "first item". This variable is setting the time at which the player will start at initially, i.e. the time offset for the first play.

I think it's cleaner to attach this `startTime` concept to the state of the tech rather than the item being played.